### ISSUE TITLE
'pipx install twisted' gets spurious binaries

### ISSUE THREAD

**Issue by warner** [pipxproject/pipx#104]
Running `pipx install twisted` on a MacOS10.14.3 system (with Homebrew's python3.7) leads to some additional binaries being installed into `~/.local/bin/` that shouldn't be there:

```
% pipx install twisted
  installed package twisted 18.9.0, Python 3.7.2
  These binaries are now globally available
    - __init__.py
    - _twistd_unix.py
    - _twistw.py
    - cftp
    - cftp.py
    - ckeygen
    - ckeygen.py
    - conch
    - conch.py
    - htmlizer.py
    - mailmail
    - mailmail.py
    - pyhtmlizer
    - tkconch
    - tkconch.py
    - trial
    - trial.py
    - twist
    - twistd
    - twistd.py
⚠️  Note: '/Users/warner/.local/bin' is not on your PATH environment variable. These binaries will not be globally accessible until your PATH is updated. Run `pipx ensurepath` to automatically add it, or manually modify your PATH in your shell's config file (i.e. ~/.bashrc).
done! ✨ 🌟 ✨
%
```

* all those `*.py` files shouldn't have been installed, especially the `__init__.py`
* my `$PATH` actually has that directory already (and `pipx install` of other packages does not complain about it), so I wonder if whatever bug is causing the first problem might also be causing the second

I'm guessing that however pipx learns about the executables is getting confused by something unusual that Twisted might be doing. I'll try to dive into this further this weekend.

thanks for the great project!
 -Brian

**Comment by cs01** [comment]
Yikes that's not good. Thanks for reporting! 

If you end up testing/finding root cause/fixing that would be amazing. 

The way to test this is by creating a virtual environment, running `pip install twisted`, then running `python venv_metadata_inspectory.py twisted <path to venv bin dir>` (located at https://github.com/pipxproject/pipx/blob/master/pipx/venv_metadata_inspector.py).i

**Comment by warner** [comment]
Hmm.

```
% python /usr/local/lib/python3.7/site-packages/pipx/venv_metadata_inspector.py twisted /tmp/v/bin
{"binaries": ["__init__.py", "_twistd_unix.py", "_twistw.py", "cftp", "cftp.py", "ckeygen", "ckeygen.py", "conch", "conch.py", "htmlizer.py", "mailmail", "mailmail.py", "pyhtmlizer", "tkconch", "tkconch.py", "trial", "trial.py", "twist", "twistd", "twistd.py"], "binary_paths": ["/tmp/v/bin/__init__.py", "/tmp/v/bin/_twistd_unix.py", "/tmp/v/bin/_twistw.py", "/tmp/v/bin/cftp", "/tmp/v/bin/cftp.py", "/tmp/v/bin/ckeygen", "/tmp/v/bin/ckeygen.py", "/tmp/v/bin/conch", "/tmp/v/bin/conch.py", "/tmp/v/bin/htmlizer.py", "/tmp/v/bin/mailmail", "/tmp/v/bin/mailmail.py", "/tmp/v/bin/pyhtmlizer", "/tmp/v/bin/tkconch", "/tmp/v/bin/tkconch.py", "/tmp/v/bin/trial", "/tmp/v/bin/trial.py", "/tmp/v/bin/twist", "/tmp/v/bin/twistd", "/tmp/v/bin/twistd.py"], "binary_paths_of_dependencies": {"setuptools": ["/tmp/v/bin/easy_install", "/tmp/v/bin/easy_install-3.6", "/tmp/v/bin/easy_install-3.7"], "Automat>=0.3.0": ["/tmp/v/bin/automat-visualize"]}, "package_version": "18.9.0", "python_version": "Python 3.7.2"}

% ls /tmp/v/bin/
activate		automat-visualize*	easy_install*		pip3*			python-config*		trial*
activate.csh		cftp*			easy_install-3.7*	pip3.7*			python3@		twist*
activate.fish		ckeygen*		mailmail*		pyhtmlizer*		python3.7*		twistd*
activate_this.py	conch*			pip*			python@			tkconch*		wheel*

```

**Comment by cs01** [comment]
Yeah something is wrong there. All the logic is in `venv_metadata_inspector.py`.

**Comment by warner** [comment]
cool, I'll dig in

**Comment by warner** [comment]
Hm, the same problem occurs with `pipx install pyflakes`.

**Comment by warner** [comment]
pyflakes has a module named `pyflakes.scripts`.. I think the `venv_metadata_inspector.py` logic that looks at RECORD and does `if path.parent.name == "scripts" in entry or path.parent.samefile(bin_path)` is getting confused by that (line 49).

I'm not even sure what that condition is doing.. having both `==` and `in` seems weird to me.

**Comment by warner** [comment]
It looks like that line has been there since the beginning. I'm not sure what the precedence rules are: `path.parent.name == "scripts" in entry` when `path.parent.name = "scripts"` and `entry = "pyflakes/scripts/__init__.py"` yields True, which is what's causing the spurious binaries. But it doesn't behave like `(path.parent.name == "scripts") in entry` (which throws a TypeError because `in (string)` doesn't take a bool as the left operand), nor does it behave like `path.parent.name == ("scripts" in entry)` (which is False because the left side is a string and the right side is a bool).

What is that line supposed to be doing?

**Comment by cs01** [comment]
Oh wow that looks like a typo! I tried running again without `path.parent.name == "scripts" in entry or` and it seemed to work fine.

**Comment by cs01** [comment]
This has been fixed in 0.12.3.3. Thanks for looking in to the source and sorry for the confusion.

```
>> pipx --version
0.12.3.3
>> pipx install twisted
  installed package twisted 18.9.0, Python 3.6.7
  These binaries are now globally available
    - cftp
    - ckeygen
    - conch
    - mailmail
    - pyhtmlizer
    - tkconch
    - trial
    - twist
    - twistd
done! ✨ 🌟 ✨
```

**Comment by warner** [comment]
thank you!
